### PR TITLE
 login with google

### PR DESCRIPTION
**Pull request by Nghicv**

### CODE REVIEW

**Comment by duyth1993** [comment]
oK

File: .gitignore [hunk 1]
```diff
@@ -24,3 +24,6 @@ config/settings/*.local.yml
 config/environments/*.local.yml
 .DS_Store
 /.idea
+
+# Ignore application configuration
+/config/application.yml
```
Comment (hellovietnam93) [inline .gitignore]: sao lai phai ignore thế

File: app/models/user.rb [hunk 2]
```diff
@@ -1,9 +1,29 @@
...
-    :validatable
+    :validatable, :omniauthable
   has_many :exams, dependent: :destroy
   has_many :questions, dependent: :destroy
   enum role: [:member, :admin]
   validates :name, presence: true
-  validates :chatwork_id, presence: true
+
+  class << self
+    def find_for_google_oauth2 access_token, signed_in_resource=nil
```
Comment (hellovietnam93) [inline app/models/user.rb]: ``` signed_in_resource = nil ```

File: app/models/user.rb [hunk 3]
```diff
@@ -1,9 +1,29 @@
...
+      data = access_token.info
+      user = User.where(provider: access_token.provider, uid: access_token.uid).first
+      if user
+        user
+      else
+        registered_user = User.where(email: access_token.info.email).first
```
Comment (hellovietnam93) [inline app/models/user.rb]: dùng find_by đi

File: app/models/user.rb [hunk 4]
```diff
@@ -1,9 +1,29 @@
...
   has_many :exams, dependent: :destroy
   has_many :questions, dependent: :destroy
   enum role: [:member, :admin]
   validates :name, presence: true
-  validates :chatwork_id, presence: true
+
+  class << self
+    def find_for_google_oauth2 access_token, signed_in_resource=nil
+      data = access_token.info
+      user = User.where(provider: access_token.provider, uid: access_token.uid).first
```
Comment (hellovietnam93) [inline app/models/user.rb]: find_by

File: .gitignore (hunk 1 above)
Comment (Nghicv) [inline .gitignore]: e dung gem figaro anh ak

**Comment by hellovietnam93** [comment]
OK

**Pull request merged**